### PR TITLE
VLTCLT-51 Add client method to get or create encryption key id

### PR DESCRIPTION
**Pull request by nicolas2bert**

### CODE REVIEW

**Comment by nicolas2bert** [comment]
ping

File: lib/IAMClient.js [hunk 1]
```diff
@@ -979,6 +979,40 @@ class VaultClient {
...
+     */
+    getOrCreateEncryptionKeyId(canonicalId, options, callback) {
+        assert(canonicalId !== undefined, 'canonicalId need to be specified');
+        assert(typeof canonicalId === 'string', 'canonicalId should be a string');
+        assert(canonicalId !== '', 'canonicalId should not be empty string');
+        const data = {
+            Action: 'GetOrCreateEncryptionKeyId',
+            canonicalId,
+        };
+        this.request('POST', '/', false, (err, data, code) => {
```
Comment (dvasilas) [inline lib/IAMClient.js:1002]: Is a POST to / doing what we want to do here? 🤔

Comment (nicolas2bert) [reply lib/IAMClient.js:1002]: Yes, :) It uses an RPC (remote procedure call) protocol. All actions are sent to a single endpoint `'/'` and the payload specifies the action to do:
```
const data = {
      Action: 'GetOrCreateEncryptionKeyId',
      canonicalId,
  };
```

Comment (dvasilas) [reply lib/IAMClient.js:1002]: Ooh I see. I'm not familiar with the vault API and I assumed it uses HTTP routes.

Comment (BourgoisMickael) [reply lib/IAMClient.js:1002]: This action doesn't exist for now I think

File: lib/IAMClient.js [hunk 2]
```diff
@@ -979,6 +979,40 @@ class VaultClient {
...
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, {
+                message: {
+                    body: data,
+                    code,
+                    message: 'Attribute retrieved or created',
```
Comment (BourgoisMickael) [inline lib/IAMClient.js]:
```suggestion
                    message: 'Encryption key retrieved or created',
```

**Review by BourgoisMickael — APPROVED** [review]

**Review by dvasilas — APPROVED** [review]

**Comment by nicolas2bert** [comment]
@bert-e approve

**Comment by nicolas2bert** [comment]
@bert-e reset

**Comment by nicolas2bert** [comment]
ping

**Comment by nicolas2bert** [comment]
@bert-e reset

**Comment by nicolas2bert** [comment]
ping

**Comment by nicolas2bert** [comment]
@bert-e reset

**Pull request merged**